.github/workflows/ci-tk.yaml

```diff
@@ -49,7 +49,7 @@ jobs:
           echo "$VENV_DIR/bin" >> "$GITHUB_PATH"
 
       - name: Install pip deps
-        if: "!contains(matrix.os, 'amdgpu') && !cancelled()"
+        if: "!contains(matrix.os, 'gpu') && !cancelled()"
```

Comment:
If we ever want to run these tests on GPUs that aren't AMD GPUs, I'd prefer to use a more specific check here.

Please restructure the `matrix` above to be explicit about which types of tests should run on which 'os' (that should really be 'runner' or 'runs-on', not 'os'). Maybe something like
```yml
      matrix:
        include:
          - name: cpu
            runs-on: ubuntu-24.04
          - name: amdgpu_mi300
            runs-on: linux-mi300-1gpu-ossci-iree-org
          - name: amdgpu_mi250
            runs-on: nodai-amdgpu-mi250-x86-64
```

Then check `matrix.name` below using `amdgpu` or `mi300`/`mi250`. Or you could get more explicit with a field for the enabled test types instead of encoding that information in the name.